### DISCUSSION TITLE
Videos with alternative file type Field

### DISCUSSION THREAD

**Discussion by JohannesMerkt** [directus/directus#8438] [Feature Requests]
When I embed a Video on a Website I create a version using WebM and one using MP4. Since WebMs are much smaller with better quality than MP4s, but not supported by every browser. When embedding the video i can then provide both alternatives and the browser can select the first it supports. Like this:
``` HTML
<video>
    <source src="name.webm" type="video/webm"/>
    <source src="name.mp4" type="video/mp4"/>
</video>
```

Thats why it would be nice to have the ability to define alternative files with different formats on videos when uploading. This might also be interesting for other file types like audio? Not sure.

**Comment by rijkvanzanten** [comment]
Interesting! So a single directus_files record would point to multiple "actual" files on disk? 🤔

**Reply by JohannesMerkt** [reply]
Thats the best solution I can think of. Maybe there are better ones. I just wanted to make the developers aware of this kind of usecase. And I do feel like its important for a well performing website to host the smallest file possible.

**Reply by benhaynes** [reply]
Seems similar to images -> thumbnails... 🤔

**Reply by joselcvarela** [reply]
I don't know if this would be a great thing to have. I imagine the files on Directus like files on a folder. Each item should represent a single file. Even though they could represent the same thing for the end user, they are clearly different.

My suggestion is, create a relationship between your model and the directus_files to put this alternative resources. So you would ended up with A <-> B <-> directus_files

It would be handy to have a preview on A, but I think it's not possible right now.

**Reply by benhaynes** [reply]
We could look into adding a junction table from Directus Files to Directus Files, allowing relating them together. Though, this is only useful if the relationship grouping can be automated (to much work to do manually).

Another idea would be adding a parent field to Directus Files. You could then nest any file within another file organizationally, while still keeping them separate. As opposed to the "related files" method above, this would create more of a hierarchy. I think I prefer this approach if we were to do anything.

**Reply by rijkvanzanten** [reply]
to take it back up a notch to the OP, it sounds like the real underlying feature request here is to allow the `format` query param on _videos_ instead of just images 😄

**Reply by benhaynes** [reply]
That too, but similar to thumbnails, any transformations created are still not formally managed by the database. One idea would be to allow (selectively) "promoting" thumbnails or video transcodings into actual items within `directus_files`, and using a `.parent` field to nest them within the source.

This means you can explicitly link a thumbnail or video format relationally, instead of using the parent and needing to add the transform params in the project code. A bigger change, but seems like a nice approach to this.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
